src/lib/patch-express-async-support.js

```diff
@@ -0,0 +1,41 @@
+const Layer = require('express/lib/router/layer');
+const Router = require('express/lib/router');
+const originalParam = Router.param;
+
+exports.init = () => {
+	Layer.prototype.__defineGetter__('handle', function () {
+		return this.__handle;
+	});
+	Layer.prototype.__defineSetter__('handle', function (fn) {
+		this.__handle = wrap(fn);
+	});
+	Router.param = param;
+};
+
+function param (...args) {
+	const processedArgs = args.map(arg => {
+		if (typeof arg !== 'function') return arg;
+		return function (req, res, next, id, name) {
+			const result = arg(req, res, next, id, name);
+			if (result && result.catch) result.catch(next);
```

Comment:
should we use `result instanceof Promise` here, or check that `result.catch` is in fact a function before calling it or something?